### PR TITLE
feat: Allow annotation GrpcClient for target type 'annotation'

### PR DESCRIPTION
**Pull request by 313hemant313**
Allow annotation GrpcClient for target type 'annotation'. #991

### CODE REVIEW

**Review by ST-DDT — CHANGES_REQUESTED** [review]
Please add a test case for this.

**Comment by 313hemant313** [comment]
@ST-DDT Test case added, please review.

File: grpc-client-spring-boot-autoconfigure/src/main/java/net/devh/boot/grpc/client/inject/metaannotation/GrpcClientWrapper.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import net.devh.boot.grpc.client.inject.GrpcClient;
+import org.springframework.core.annotation.AliasFor;
+
+@GrpcClient(value = "")
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.FIELD, ElementType.METHOD})
+public @interface GrpcClientWrapper {
```
Comment (ST-DDT) [inline grpc-client-spring-boot-autoconfigure/src/main/java/net/devh/boot/grpc/client/inject/metaannotation/GrpcClientWrapper.java:15]: AFAICT this is only for the test isn't it? Please move it to the tests.

Comment (313hemant313) [reply grpc-client-spring-boot-autoconfigure/src/main/java/net/devh/boot/grpc/client/inject/metaannotation/GrpcClientWrapper.java]: Correct, my bad. Moved to test.

**Comment by ST-DDT** [comment]
CI is red.

**Comment by 313hemant313** [comment]
Updated after ./gradlew :tests:spotlessApply. please run the job.

**Comment by ST-DDT** [comment]
Please pull and run the `./gradlew spotlessApply` command again.

**Comment by 313hemant313** [comment]
./gradlew check and  spotlessApply are success now.

**Comment by 313hemant313** [comment]
I was using java 19 version. changed it to 17 and pushed a new change with ./gradlew :tests:spotlessApply

**Review by ST-DDT — APPROVED** [review]
Looks good to me. Thanks for your contribution. ❤️ 

FFR: I find PRs easier to review if they aren't force pushed in between.

**Comment by 313hemant313** [comment]
> FFR: I find PRs easier to review if they aren't force pushed in between.
Okay.

**Comment by ST-DDT** [comment]
> How to get this merged ?

You cannot merge this. We will have another reviewer look at this and then merge it ourselves.

**Review by yidongnan — APPROVED** [review]
LGTM, Thanks for your contribution.

**Pull request merged**